api/v1alpha1/provisioning_types.go

```diff
@@ -84,7 +94,7 @@ type ProvisioningSpec struct {
 	// installation. If using metal3 for power management, BMCs must be
 	// accessible from the machine networks. User should provide two IPs on
 	// the external network that would be used for provisioning services.
-	ProvisioningNetwork string `json:"provisioningNetwork,omitempty"`
+	ProvisioningNetwork ProvisioningNetwork `json:"provisioningNetwork,omitempty"`
 }
```

Comment:
Here the type of ProvisioningNetwork has changed but in the resulting yaml file it is still a string.